src/Blueprint.h

```diff
@@ -349,23 +346,106 @@ namespace snowcrash {
     /** Collection of Resources */
     typedef Collection<Resource>::type Resources;
 
-    /**
-     *  Group of API Resources
-     */
-    struct ResourceGroup {
+    /** Forward declaration of Element */
+    struct Element;
 
-        /** A Group Name */
-        Name name;
+    /** Collection of elements */
+    typedef std::vector<Element> Elements;
 
-        /** Group description */
-        Description description;
+    /** Element */
+    struct Element {
+
+        /** Class of an element */
+        enum Class {
+            UndefinedElement = 0, // Unknown
+            CategoryElement,      // Group of other elements
+            CopyElement,          // Human readable text
+            ResourceElement,      // Resource
+            DataStructureElement  // Data Structure
+        };
+
+        /** Attributes of an element */
+        struct Attributes {
+
+            /** Human readable name of the element */
+            Name name;
+        };
+
+        /** Content of an element */
+        struct Content {
+
+            /** EITHER Copy */
+            std::string copy;
+
+            /** OR Resource */
+            Resource resource;
+
+            /** OR Data Structure */
+            DataStructure dataStructure;
+
+            /** OR Collection of elements */
+            Elements& elements();
+            const Elements& elements() const;
+
+            /** Constructor */
+            Content();
+
+            /** Copy constructor */
+            Content(const Element::Content& rhs);
+
+            /** Assignment operator */
+            Content& operator=(const Element::Content& rhs);
+
+            /** Destructor */
+            ~Content();
 
-        /** Resources */
-        Resources resources;
+        private:
+            std::auto_ptr<Elements> m_elements;
+        };
+
+        /** Type of Category element (parser internal flag) */
+        enum Category {
+            UndefinedCategory = 0,     // Unknown
+            ResourceGroupCategory,     // Resource Group
+            DataStructureGroupCategory // Data Structure Group
+        };
+
+        /** Type of the element */
+        Element::Class element;
+
+        /** Attributes of the element */
+        Element::Attributes attributes;
+
+        /** Content of the element */
+        Element::Content content;
+
+        /** Type of Category element (to be used internally only) */
+        Element::Category category;
+
+        /** Constructor */
+        Element(const Element::Class& element_ = Element::UndefinedElement);
+
+        /** Copy constructor */
+        Element(const Element& rhs);
+
+        /** Assignment operator */
+        Element& operator=(const Element& rhs);
+
+        /** Destructor */
+        ~Element();
+    };
+
+    /**
+     *  Group of API Resources (Category Element)
+     */
+    struct ResourceGroup : public Element {
```

Comment:
Due to our architecture, just `typedef`ing them won't work. We need both `SectionProcessor<ResourceGroup>` and `SectionProcessor<DataStructureGroup>`, but using `typedef` will give redefinition errors.